sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1289,6 +1291,12 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
         // Create a project if this is an INSERT INTO BY NAME query.
         val projectByName = if (i.userSpecifiedCols.nonEmpty) {
           Some(createProjectForByNameQuery(i))
+        } else if (i.byName) {
+          val partition = i.partitionSpec.filter(p => p._2.isDefined).keySet
+          val output = i.table.output.filter(col => {
+            !partition.exists(p => conf.resolver(col.name, p))
+          })
+          Some(resolveOutputColumns(r.table.name, output, i.query, i.byName, conf))
```

Comment:
why do we reorder columns here? It will be handled by `ResolveOutputRelation` rule later, after we turn `InsertIntoStatement` into v2 write command.

Reply:
Thanks for remind, removed.